### ISSUE TITLE
registerForRemoteNotifications

### ISSUE THREAD

**Issue by benoist** [evollu/react-native-fcm#44]
Hi,

I'm struggling to get the remote push notifications working on IOS.
Reading through the documentation I found that registerForRemoteNotifications should be called to allow remote push notifications to be received. However in the code I only see registerUserNotificationSettings being called.
I also read that on macOS you only need to call that method and it will call the registerForRemoteNotifications for you, however on iOS it seems you'd have to explicitly call that method for remote push notifications to work.

Am I missing something in the documentation or should this method be called?

**Comment by gorangajic** [comment]
you register for notifications using 
```
    import FCM from 'react-native-fcm';
    FCM.requestPermissions();
    FCM.getFCMToken().then(token => {
      console.log(token);
    });
```
and then you can send notification with firebase api using that token.

btw push notifications only works on the real device

**Comment by benoist** [comment]
I have done all that. I can receive the notification when the app is on foreground, but not in the background.
I've read the other issues surrounding this and used the notification payload with the high priority, however the remote notifications are not showing up.

I've compared the requestPermissions function in the FCM code to the GCM code:

https://github.com/evollu/react-native-fcm/blob/master/ios/RNFIRMesssaging.m#L102
https://github.com/evollu/react-native-gcm-push-notification/blob/master/ios/RNGCM.m#L129

as you can see the `registerForRemoteNotifications`call is missing in FCM.

**Comment by benoist** [comment]
I will check later today if calling the `registerForRemoteNotifications` method helps and then post back here.

**Comment by evollu** [comment]
you don't have to add that method because fire base SDK does this for you automatically.
can you try to use firebase native sample project to see if you can receive notification? want to make sure it is related to this repo or now

**Comment by benoist** [comment]
Ok so did some tests

**Removed old app, run on phone with react-native-fcm**

1. Set app to background
2. Send notification with high priority
3. Did not receive notification
4. Open app, receive notification in console from FCM

**Removed old app, run on phone with react-native-fcm + registerForRemoteNotifications**

1. Set app to background
2. Send notification with high priority
3. Instantly received notification in ios notification center

Also the firebase documentation shows these lines when registering the notification settings:

```Objective-c
UIUserNotificationType allNotificationTypes =
(UIUserNotificationTypeSound | UIUserNotificationTypeAlert | UIUserNotificationTypeBadge);
UIUserNotificationSettings *settings =
[UIUserNotificationSettings settingsForTypes:allNotificationTypes categories:nil];
[[UIApplication sharedApplication] registerUserNotificationSettings:settings];
[[UIApplication sharedApplication] registerForRemoteNotifications];
```

When I remove all react native fcm code from the repo and use the examples from firebase, the notifications come instantly as well.

**Comment by evollu** [comment]
@benoist released. Thank you

**Comment by benoist** [comment]
@evollu Thank you for the quick release!

**Comment by umhan35** [comment]
Hello, a user of my app emailed me complaining not receiving notifications. But I tested on an iPad device and it did receive push notifications instantly.

I'm using 0.29 and `[[UIApplication sharedApplication] registerForRemoteNotifications];` is not in my production app. Is this the reason why the user did not receive notifications? But why, missing this line of code, I also receive instant notifications on development phone and normal phone?

**Comment by evollu** [comment]
I assume it works for some particular OS versions. try the latest version and check if the user can receive notifications after that

**Comment by umhan35** [comment]
Apple shouldn't allow it work... The user is using iOS 10, which is probably the version that missing `registerForRemoteNotifications ` won't work. But without it, notifications works on 9.3.2.

**Comment by evollu** [comment]
my 9.3.2 is working fine too...anyway let me know if the fix solves the problem

**Comment by umhan35** [comment]
@benoist What iOS version you are using that push notifications are not working?

**Comment by benoist** [comment]
I'm using 9.3.2

**Comment by benoist** [comment]
I might be able to explain by the use of my findings.
If you had the app already installed in a pre iOS 8 version, setting up remote notifications worked differently. If you never delete the app, the registration for remote notifications remains. What I noticed was that after using that line I was able to receive notifications. I could then remove the line and update the app without deleting first and still receive notifications. Only after deleting the app, the registerForRemoteNotfications is required again.
